openweathermap-ios/openweathermap-ios/BookmarkLocation/Manager/BookmarkLocationManager.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  BookmarkLocationService.swift
+//  openweathermap-ios
+//
+//  Created by Cassio Sousa on 24/04/20.
+//
+
+import Foundation
+
+class BookmarkLocationManager: NSObject {
+
+    func save(bookmarkLocation: BookmarkLocation) -> [BookmarkLocation] {
+        var allBookmarks = getAll()
+        let defaults = UserDefaults.standard
+        allBookmarks.append(bookmarkLocation)
+        let archiveData = self.archiveData(allBookmarks)
+        defaults.set(archiveData, forKey: BookmarLocationEnumaration.bookmarkLocation.rawValue)
+        return allBookmarks
+    }
+
+    func getAll() -> [BookmarkLocation] {
+        return unArchiveData()
+    }
+
+    fileprivate func archiveData(_ allBokkmarks: [BookmarkLocation]) -> Data {
+        do {
+            let archiveData = try NSKeyedArchiver.archivedData(withRootObject: allBokkmarks, requiringSecureCoding: false)
+            return archiveData
+        } catch let error {
+            debugPrint(error)
+        }
+        return Data()
```

Comment:
You cannot use Dummy data. The client of this class has to know that the operation was not succeded. You cannot hide the error from him.

Reply:
Fixed in b6bb083